catroid/src/main/java/org/catrobat/catroid/ProjectManager.java

```diff
@@ -304,6 +306,36 @@ private static void loadLegoEV3SettingsFromProject(Project project, Context cont
 		}
 	}
 
+	public List<UserVariable> getGlobalVariableConflicts(Project project1, Project project2) {
+		List<UserVariable> project1GlobalVars = project1.getUserVariables();
+		List<UserVariable> project2GlobalVars = project2.getUserVariables();
+		List<UserVariable> conflicts = new ArrayList<>();
+		for (UserVariable project1GlobalVar : project1GlobalVars) {
+			for (UserVariable project2GlobalVar : project2GlobalVars) {
+				if (project1GlobalVar.getName().equals(project2GlobalVar.getName()) && !project1GlobalVar.getValue()
+						.equals(project2GlobalVar.getValue())) {
+					conflicts.add(project1GlobalVar);
+				}
+			}
+		}
+		return conflicts;
+	}
+
+	public List<UserList> getGlobalListConflicts(Project project1, Project project2) {
+		List<UserList> project1GlobalLists = project1.getUserLists();
+		List<UserList> project2GlobalLists = project2.getUserLists();
+		List<UserList> conflicts = new ArrayList<>();
+		for (UserList project1GlobalList : project1GlobalLists) {
+			for (UserList project2GlobalList : project2GlobalLists) {
+				if (project1GlobalList.getName().equals(project2GlobalList.getName()) && !project1GlobalList.getList()
+						.equals(project2GlobalList.getList())) {
```

Comment:
Be careful: UserList.equals does only check the type and name of the user list itself. It does not check every item with the value.
I think the equals function should be fixed first, but then there will emerge side effects. 
maybe write an extra equals which checks every item in the list or better fix the equals?

Reply:
Hi, so I was using project1GlobalList.getList() which will give me an ArrayList to check if each element is equal in the code. I think that should work?

Reply:
ok, right, the ArrayList.equals should do it